pykafka/connection.py

```diff
@@ -63,7 +76,8 @@ def connect(self, timeout):
         """Connect to the broker."""
         self._socket = socket.create_connection(
             (self.host, self.port),
-            timeout=timeout / 1000
+            timeout / 1000,
+            self.source_host, self.source_port
```

Comment:
I was afraid that was the case. The stdlib docs are super confusing about that sometimes and functions will sometimes accept the kwarg when they shouldn't.